src/ocl/convolutionocl.cpp

```diff
@@ -1591,6 +1540,164 @@ static inline bool IsAlgorithmDisabled(const miopenConvAlgorithm_t algo)
+
+    auto ctx = ConvolutionContext{problem};
+    ctx.SetStream(&handle);
+    ctx.DetectRocm();
+
+    const auto& map = miopen::solver::GetMapValueToAnySolver();
+    for(const auto& item : map)
+    {
+        const auto solver_id = solver::Id{item.first};
+        // solver_id is always valid here, because taken from registry.
+        // Validity check is not required.
+        const auto algo = solver_id.GetAlgo();
+        if(IsAlgorithmDisabled(algo)) // Algos can be disabled globally.
+            continue;
+        const auto& s = item.second;
+        if(!s.IsApplicable(ctx))
+            continue;
+        if(!s.IsDynamic()) // Let's allow non-dynamic later, if necessary.
+        {
+            MIOPEN_LOG_I2(solver_id.ToString() << " Not dynamic, skipped");
+            continue;
+        }
+
+        // gemm can appear here only after actual (non-dummy) GEMM Solver is implemented.
+        if(solver_id == solver::Id::gemm())
+            MIOPEN_LOG_W("GEMM solver is ready, rework this function");
+
+        const auto wti = s.GetWti(ctx);
+        MIOPEN_LOG_I2(solver_id.ToString() << " Estimated WTI = " << wti);
+        if(wti <= 0.0f) // Skip unknown WTIs and avoid DIV/0.
+            continue;
+        const auto time = 10.0f / wti; // Assume WTI == 1.0 (100%) is 10 ms.
+
+        interim.emplace_back(time, s.GetWorkspaceSize(ctx), solver_id.Value(), algo);
+    }
+
+    // Dual purpose variable:
+    // * Used as index for writing into output array (solutions).
+    // * Counts the number of entries written, yielding value for solutionsCount.
+    auto i = std::size_t{0};
+
+    if(!interim.empty())
+    {
+        std::sort(begin(interim), end(interim));
+        for(const auto& entry : interim)
+        {
+            if(i >= maxSolutionCount)
+                break;
+            if(solutions != nullptr)
+                solutions[i] = entry;
+            ++i;
+        }
+        *solutionCount = i;
+        /// Right now we do not have GetWti() for GEMM.
+        /// And only those solutions that are faster than GEMM return WTI.
+        /// Therefore it is Ok for now to not use GEMM if some other solution is found.
+        /// \todo Rework this when we have GetWti() for GEMM.
+        return;
+    }
+
+    /// Separate path for GEMM algo, intermediate implementation.
+    /// \todo Remove when GEMM Solver(s) ready.
+    if(i >= maxSolutionCount)
+    {
+        // Do nothing.
+    }
+    else if(problem.direction.IsForward())
+    {
+        if(IsGemmApplicableFwd(weightsDesc, inDesc, outDesc) && (i < maxSolutionCount))
+        {
+            if(solutions != nullptr)
+            {
+                solutions[i].algorithm = miopenConvolutionAlgoGEMM;
+                solutions[i].time      = -1.0;
+                solutions[i].workspace_size =
+                    ForwardGetValidWorkSpaceSizeGemm(handle, weightsDesc, inDesc, outDesc);
+                solutions[i].solution_id = solver::Id::gemm().Value();
+            }
```

Comment:
We cant assume that GEMM always performs poorly. In many cases GEMM is better than other kernels. Unless this is temporary, in which case it is fine.

Reply:
This is temporary code. I am going to implement approximate GetWti for GEMM when GEMM solver is ready.